### ISSUE TITLE
Running Rancher Server 2 as part of an Istio mesh

### ISSUE THREAD

**Issue by moshe0076** [rancher/rancher#15787]
**Rancher versions:**
rancher/server or rancher/rancher:
server V2.0.8
rancher/agent or rancher/rancher-agent:

**Infrastructure Stack versions:**
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable):
EKS 1.10

**Docker version: (`docker version`,`docker info` preferred)**

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**

**Steps to Reproduce:**
I want to run the Rancher server as part of the Istio mesh. I have an SSL termination on the ALB but the rancher server, which receives the initiated connection from the ALB via the Envoy sidecar on HTTP redirects to HTTPS which results in an endless loop of redirects.

Is there a way to make the rancher server not redirect or just work on HTTP?

Thanks!

Moshe

**Results:**

**Comment by Oats87** [comment]
Rancher will not force a redirect if the following header is set:
```
X-FORWARDED-PROTO: https
```

More information about this can be found here: https://rancher.com/docs/rancher/v2.x/en/installation/ha/rke-add-on/layer-7-lb/#2-configure-load-balancer

**Comment by moshe0076** [comment]
I think the problem is that the Envoy is passing X-Forwarded-Proto: http even if the connection to the ALB was HTTPS. Is there a flag to signal the Rancher server not to force-redirect to HTTPS even if the X-Forwarded-Proto header is set to HTTP? 

There may be some workaround by using Istio VirtualService (https://github.com/istio/istio/issues/7964).

Thanks!

**Comment by loganhz** [comment]
No, I guess. Reopen this issue. Maybe we need to add a flag for that

**Comment by xeor** [comment]
Is this still a problem? I'm trying to install rancher when istio is enabled, but end up in a redirect loop.
I tried setting `X-Forwarded-Proto=https`

**Comment by lukasz-bielinski** [comment]
Hi @moshe0076  did you managed to run rancher as part of istio mesh? especially newer versions of rancher e.g. 2.7.x line. I noticed so far problems with wss connections in multi replica setup
